devito/passes/clusters/utils.py

```diff
@@ -0,0 +1,104 @@
+from collections import Iterable, OrderedDict
+from functools import wraps
+
+from devito.symbolics import retrieve_terminals
+from devito.tools import flatten, timed_pass
+from devito.types import Dimension, Symbol
+
+__all__ = ['dse_pass', 'makeit_ssa', 'make_is_time_invariant']
+
+
+def dse_pass(func):
+    @wraps(func)
+    def wrapper(*args):
+        if timed_pass.is_enabled():
+            maybe_timed = lambda *_args: timed_pass(func, func.__name__)(*_args)
+        else:
+            maybe_timed = lambda *_args: func(*_args)
+        args = list(args)
```

Comment:
Probably irrelevant here so please ignore if `args` will always be such that it nevermatters, but safer to use `as_list`?